gtsam/hybrid/DCSAM.h

```diff
@@ -0,0 +1,202 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file    DCSAM.h
+ * @brief   Discrete-Continuous Smoothing and Mapping for Factored Models
+ * @author  Kevin Doherty
+ * @author  Varun Agrawal
+ * @date    March 2025
+ */
+
+#pragma once
+
+#include <gtsam/discrete/DiscreteFactor.h>
+#include <gtsam/discrete/DiscreteFactorGraph.h>
+#include <gtsam/discrete/DiscreteKey.h>
+#include <gtsam/discrete/DiscreteMarginals.h>
+#include <gtsam/hybrid/HybridNonlinearFactorGraph.h>
+#include <gtsam/inference/FactorGraph.h>
+#include <gtsam/inference/Symbol.h>
+#include <gtsam/nonlinear/ISAM2.h>
+#include <gtsam/nonlinear/Marginals.h>
+#include <gtsam/nonlinear/NonlinearFactorGraph.h>
+
+#include <map>
+#include <set>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+namespace gtsam {
+
+class GTSAM_EXPORT DCSAM {
+ private:
+  // Global factor graph and iSAM2 instance
+  NonlinearFactorGraph nfg_;
+  DiscreteFactorGraph dfg_;
+  ISAM2 isam_;
```

Comment:
I always wanted to be able to configure what solver you used here. In principle you should be able to use any continuous solver you want (and any discrete solver, really). Would be a cool feature in the future.

Reply:
True! I'll add a todo